python/lsst/daf/butler/core/datasets.py

```diff
@@ -269,3 +281,15 @@ def detach(self):
         ref = deepcopy(self)
         ref._id = None
         return ref
+
+    def lookupNames(self):
+        """Names to use when looking up this DatasetRef in a configuration.
+
+        The names are returned in order of priority.
+
+        Returns
+        -------
+        names : `tuple` of `str`
+            Tuple of the `DatasetType` name and the `StorageClass` name.
+        """
+        return self.datasetType.lookupNames()
```

Comment:
Why are there multiple names? Why is there a lookup in configuration for a specific `DatasetRef`? If the intent is to lookup something about the `DatasetType` in the configuration i'd rather explicitly have the user do `datasetRef.datasetType.lookupNames()` such that the intention is clear.

Reply:
Doing it like this means I have a single interface for DatasetType, DatasetRef and StorageClass so I call one method on whatever I've been given and I can look it up in the config. Without this I have to have an explicit `isinstance` check in those cases and forward it on myself. Since we want all these look ups to work with all three types it seemed easier and clearer to do it this way. Whilst looking at this I also realized that in theory we can have a disagreement between `DatasetRef.components` and `DatasetRef.datasetType.storageClass.components` and we never check.